### PR TITLE
Fix #785: add squash support

### LINKED ISSUE #785: Question: Is docker optimise similar to what docker-squash does?
I'm trying to reduce the size of my docker image. Is setting optimise to true similar to what docker-squash does?

Would there be a squash option in the future?

### PR DESCRIPTION
**Pull request by dexterhu**
* Allow to squash docker images as option in pom.xml, e.g.:
```
 <build>
          <assembly>
                    <descriptorRef>artifact-with-dependencies</descriptorRef>
           </assembly>
           <optimise>true</optimise>
           <compression>gzip</compression>
           <squash>true</squash>
</build>
```
* Default is disabled, users need to enable it at docker daemon
https://docs-stage.docker.com/engine/reference/commandline/dockerd/#daemon-configuration-file
```
{
  "experimental": true
}
```
https://docs-stage.docker.com/engine/reference/commandline/image_build/
https://docs-stage.docker.com/engine/reference/commandline/dockerd/#daemon-configuration-file

### CODE REVIEW

**Comment by dexterhu** [comment]
tested with a SNAPSHOT release at https://github.com/dexterhu/docker-maven-plugin/packages/340297?version=0.34-SNAPSHOT

**Comment by rhuss** [comment]
Thanks ! I will try to have a look after my PTO (which starts now) in 2 weeks ...

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
   <groupId>io.fabric8</groupId>
   <artifactId>docker-maven-plugin</artifactId>
-  <version>0.33-SNAPSHOT</version>
+  <version>0.34-SNAPSHOT</version>
```
Comment (rhuss) [inline pom.xml:9]: we are still on 0.33-SNAPSHOT ...

File: src/main/asciidoc/inc/external/_property_configuration.adoc [hunk 2]
```diff
@@ -229,6 +229,9 @@ when a `docker.from` or a `docker.fromExt` is set.
 | *docker.noCache*
 | Don't use a cache when building the image
 
+| *docker.squash*
+| Squash newly built layers into a single new layer (API 1.25+, need to enable it in Daemon configuration file)
```
Comment (rhuss) [inline src/main/asciidoc/inc/external/_property_configuration.adoc:233]:
```suggestion
| Squash newly built layers into a single layer (API 1.25+, needs to be enabled in the Docker daemon configuration)
```

**Review by rhuss — CHANGES_REQUESTED** [review]
Thanks ! looks good to me, with some minor cosmetic suggestions. I think we can merge this soon.

**Comment by rhuss** [comment]
Could you please also add an entry to the changelog.md ? thanks !

File: pom.xml (hunk 1 above)
Comment (dexterhu) [reply pom.xml]: rolled back to 0.33-SNAPSHOT

File: src/main/asciidoc/inc/external/_property_configuration.adoc (hunk 2 above)
Comment (dexterhu) [reply src/main/asciidoc/inc/external/_property_configuration.adoc]: updated.

**Comment by dexterhu** [comment]
> Could you please also add an entry to the changelog.md ? thanks !

added a new item under `0.33-SNAPSHOT` in changelog.md

**Pull request merged**